opm/input/eclipse/EclipseState/Grid/FieldData.hpp

```diff
@@ -190,6 +191,41 @@ namespace Opm::Fieldprops {
                       value::status::valid_default);
         }
 
+        void default_assign_global(const std::vector<T>& src)
+        {
+            if (!global_data) {
+                throw std::invalid_argument {
+                    "Cannot call default_assign_global on keyword with local storage"
+                };
+            }
+
+            if (src.size() != this->global_data->size()) {
+                throw std::invalid_argument {
+                    "Size mismatch got: " + std::to_string(src.size()) +
+                    ", expected: " + std::to_string(this->dataSize())
+                };
+            }
+
+            std::copy(src.begin(), src.end(), this->global_data->begin());
+            std::fill(this->global_value_status->begin(), this->global_value_status->end(),
+                      value::status::valid_default);
+        }
+
+        void update_local_from_global(std::function<std::size_t(std::size_t)> local_to_global)
```

Comment:
Very strictly speaking you don't need `std::function<>` here.  We're in a header and could just as well make the function a template instead.

Reply:
I could also have passed EclipseGrid with the complete type, but I really did not want to pull in that header here. This class was initially internal to opm-common.

I doubt that there are any real performance issues due to the way it is now and might compile faster (famous last words without any proofs :grimacing:)